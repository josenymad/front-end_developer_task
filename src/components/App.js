import React, { useEffect, useState } from "react";
import "../styles/app.css";
import getMatchData from "../requests/getMatchData";
import MatchInfo from "./MatchInfo";

const App = () => {
  const [matchData, setMatchData] = useState();
  const [alert, setAlert] = useState(
    "If the stats haven't loaded after a while there may be a problem with the server",
  );

  useEffect(() => {
    getMatchData(setMatchData, setAlert);
  }, []);

  if (matchData) {
    const { name: competitionName } = matchData.meta.competition;

    return (
      <div className="App">
        <h1 className="App__header">{competitionName}</h1>
        <MatchInfo matchData={matchData} />
      </div>
    );
  }

  return (
    <p className="App__no-data">{alert}. Sorry, please try again later.</p>
  );
};

export default App;
