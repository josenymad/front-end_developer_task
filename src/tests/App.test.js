import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("Premier League", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders error message when no data is fetched", () => {
    render(<App />);
    const errorMessage = screen.getByText(
      "If the stats haven't loaded after a while there may be a problem with the server. Sorry, please try again later.",
    );

    expect(errorMessage).toBeInTheDocument();
  });
});
