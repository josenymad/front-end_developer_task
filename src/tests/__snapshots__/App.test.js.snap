// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Premier League renders correctly 1`] = `
<DocumentFragment>
  <p
    class="App__no-data"
  >
    If the stats haven't loaded after a while there may be a problem with the server. Sorry, please try again later.
  </p>
</DocumentFragment>
`;
